betterjails/src/main/java/io/github/emilyydev/betterjails/commands/CommandHandler.java

```diff
@@ -26,383 +26,317 @@
+  public CompletableFuture<Void> deleteJail(final CommandSender sender, final Jail jail) {
+    final String name = jail.name();
+    return this.plugin.jailData().removeJail(name).handleAsync((v, ex) -> {
       if (ex == null) {
-        sender.sendMessage(this.configuration.messages().deleteJailSuccess(sender.getName(), jail));
+        sender.sendMessage(this.configuration.messages().deleteJailSuccess(sender.getName(), name));
       } else {
-        LOGGER.error("An error occurred deleting data for jail {}", jail, ex);
+        LOGGER.error("An error occurred deleting data for jail {}", name, ex);
         sender.sendMessage(color("&cThere was an error while trying to remove the jail."));
       }
+
+      return null; // TODO(emilia): maybe add new Captions
     }, this.plugin);
   }
 
-  private void playerLogin(final PlayerLoginEvent event) {
-    final Player player = event.getPlayer();
-    this.namesToUuid.put(player.getName().toLowerCase(Locale.ROOT), player.getUniqueId());
+  @Permission("betterjails.betterjails")
+  @Command("betterjails")
+  @CommandDescription("Prints the version of the plugin")
+  public void printInfo(final CommandSender sender) {
+    sender.sendMessage(color("&bBetterJails &3by &bemilyy-dev &3- v%s", this.plugin.getDescription().getVersion()));
+  }
+
+  @Permission("betterjails.betterjails.reload")
+  @Command("betterjails reload")
+  @CommandDescription("Reloads the configuration file, prisoner data and jail data")
+  public void reloadData(final CommandContext<CommandSender> ctx, final CommandSender sender) {
+    try {
+      this.plugin.reload();
+      this.plugin.eventBus().post(PluginReloadEvent.class, sender);
+      sender.sendMessage(this.configuration.messages().reloadData(sender.getName()));
+    } catch (final IOException ex) {
+      LOGGER.error("An error occurred reloading plugin data", ex);
+      throw new CommandError(ctx, CommandError.RELOAD_FAILED, CommandError.executorVariable(sender.getName()));
+    }
+  }
+
+  @Permission("betterjails.betterjails.save")
+  @Command("betterjails save")
+  @CommandDescription("Saves jail and prisoner data on the spot")
+  public CompletableFuture<Void> saveData(final CommandContext<CommandSender> ctx, final CommandSender sender) {
+    return this.plugin.saveAll().<CompletableFuture<Void>>handle((v, ex) -> {
+      if (ex == null) {
+        sender.sendMessage(this.configuration.messages().saveData(sender.getName()));
+        return CompletableFuture.completedFuture(null);
+      } else {
+        // exception is logged in BJP#saveAll()
+        return CompletableFutures.failedFuture(
+            new CommandError(ctx, CommandError.SAVE_ALL_FAILED, CommandError.executorVariable(sender.getName()))
+        );
+      }
+    }).thenCompose(Function.identity());
+  }
+
+  @Parser
+  public Jail resolveJail(final CommandContext<CommandSender> ctx, final CommandInput input) {
+    final String name = input.readInput();
```

Comment:
This doesn't work, I'm gonna leave it to you to figure out why.
If I run `/jail rymiel foo 1h`, this `name` variable becomes "jail rymiel", when it should be "foo"
(the makes all jail commands fail)
(and also the prisoner ones, since it uses the same mechanism)
idk